src/map/clif.cpp

```diff
@@ -25141,25 +25141,40 @@ void clif_specialpopup(map_session_data& sd, int32 id ){
 #endif
 }
 
+static int min_tick_interval = 100;
+static int32 ban_time = min_tick_interval*10000;
+struct packet_spammer_data{
+	uint32 client_addr;
+	t_tick tick;
+	struct{
+		short count;
+		short max=10;
+	}click;
+};
+static std::vector<packet_spammer_data> packet_spammer;
+
+static bool psd_is_spammer(packet_spammer_data* psd)
+{
+	if(!psd)
+		return false;
+	if(psd->click.count < psd->click.max)
+		return false;
+
+	return true;
+}
+
 /*==========================================
  * Main client packet processing function
  *------------------------------------------*/
 static int clif_parse(int fd)
 {
 	int cmd, packet_len;
-	TBL_PC* sd;
-	int pnum;
+	int piter;
 #ifdef PACKET_OBFUSCATION
 	int cmd2;
 #endif
 
-	//TODO apply delays or disconnect based on packet throughput [FlavioJS]
-	// Note: "click masters" can do 80+ clicks in 10 seconds
-
-	for( pnum = 0; pnum < 3; ++pnum )// Limit max packets per cycle to 3 (delay packet spammers) [FlavioJS]  -- This actually aids packet spammers, but stuff like /str+ gets slow without it [Ai4rei]
```

Comment:
Why remove this? this is a performance decrease. If you really don't like it, make it configurable